### PR TITLE
improve(relayer inventory management): Match deposits and finalizations by amount

### PR DESCRIPTION
**Pull request by kevinuma**
Details: https://linear.app/uma/issue/ACX-14/monitor-should-track-l1-l2-bridged-funds-from-hubpool-to-spokepool

### CODE REVIEW

File: src/clients/bridges/BaseAdapter.ts [hunk 1]
```diff
@@ -105,36 +105,13 @@ export class BaseAdapter {
...
-        // with the most recent event last. The event set contains all deposit events that have ever happened on L1 for
-        // the relayer. We know the L1 event index where which matches to the most recent L2 finalization event
-        // (associatedL1DepositIndex). If we take a slice of the array from this index to the end then we have only events
-        // that have occurred on L1 and have no matching event on L2 (and have happened in the last day due to the
-        // previous filter). These events are deposits that must have been made on L1 but not are not yet finalized on L2.
-        const l1EventsToConsider = this.l1DepositInitiatedEvents[monitoredAddress][l1Token].slice(
-          associatedL1DepositIndex + 1
+        // Match deposits and finalizations by amount. We're only doing a limited lookback of events so collisions
+        // should be unlikely.
+        const finalizedAmounts = new Set(l2FinalizationSet.map((finalization) => finalization.amount.toString()));
```
Comment (nicholaspai) [inline src/clients/bridges/BaseAdapter.ts:110]: Are there no other `finalization` props we can use to reduce the chance of collisions more? Also, what are the implications of collisions? Does that mean that the monitor will think a deposit is finalized when it might actually be stuck? When would we ever know about the stuck situation in this case?

Comment (kevinuma) [reply src/clients/bridges/BaseAdapter.ts:110]: Unfortunately no. If there's a collision, we would mistakenly think a past transfer to have already been finalized while a new transfer for the amount to not be finalized yet. So if we set a delay before sending alerts about stuck transfers, we'd miss it if the second deposit is relatively new. But this is fine imo, as we should have caught the first deposit  (as stuck) before the second deposit happened.

In summary, from the perspective of only using this for inventory management and monitoring stuck transfers, there should be no problems.

**Comment by kevinuma** [comment]
Added tests

**Review by nicholaspai — APPROVED** [review]
